engines/order_management/app/services/order_management/order/updater.rb

```diff
@@ -157,7 +157,7 @@ def infer_payment_state
       def infer_payment_state_from_balance
         # This part added so that we don't need to override
         # order.outstanding_balance
-        balance = order.outstanding_balance
+        balance = OrderBalance.new(order).to_f
         balance = -1 * order.payment_total if canceled_and_paid_for?
```

Comment:
Will this need to be updated now that the balance will be different for cancelled orders?

Reply:
right! nicely spot @Matt-Yorkley I was blindly replacing calls to `#outstanding_balance` and missed this